R/fda-table_02.R

```diff
@@ -29,274 +25,127 @@ NULL
-          row_label2 %in% c("Median", "Min", "Max") ~ "Median (Min - Max)",
-          .default = row_label2
-        )
-      ) %>%
-      tidyr::separate_rows(c("param", "value"), sep = ";") %>%
-      mutate(value = as.numeric(value))
-
-    if (show_colcounts) {
-      colcounts <- summary(if (!is.null(alt_counts_df)) alt_counts_df[[arm_var]] else df[[arm_var]])
-      big_ns <- tibble(
-        column = c(levels(df[[arm_var]]), lbl_overall),
-        param = "bigN",
-        value = if (!is.null(lbl_overall)) c(colcounts, sum(colcounts)) else colcounts,
-      )
-      tbl <- bind_rows(tbl, big_ns)
-    }
-
-    if (prune_0) {
-      tbl <- tbl %>%
-        group_by(ord_layer_index, ord_layer_2) %>%
-        filter(sum(value) > 0) %>%
-        ungroup()
-    }
-    big_n_tbl <- if (show_colcounts) big_n_structure(param_val = "bigN", n_frmt = frmt("\n(N=xx)")) else NULL
+  # create a table using gtsummary
+  df %>%
+    gtsummary::tbl_summary(
+      by = arm_var
+    )
 
-    tbl <- tfrmt(
-      group = c(tbl_lbl, row_label1),
-      label = row_label2,
-      column = column,
-      param = param,
-      value = value,
-      title = annotations[["title"]],
-      subtitle = paste(annotations[["subtitles"]], collapse = ", "),
-      sorting_cols = c(ord_layer_1, ord_layer_2),
-      body_plan = body_plan(
-        frmt_structure(
-          group_val = ".default", label_val = ".default",
-          frmt_combine("{n} {pct}", n = frmt("xx"), pct = frmt_when("==0" ~ "", TRUE ~ frmt("(xx.x%)")))
-        ),
-        frmt_structure(
-          group_val = ".default", label_val = "Mean (SD)",
-          frmt_combine("{Mean} ({SD})", Mean = frmt("xx.x"), SD = frmt("xx.x"))
+  tbl <- df %>%
+    gtsummary::tbl_summary(
+      by = arm_var,
+      type = all_continuous() ~ "continuous2",
+      statistic = list(
+        all_continuous() ~ c(
+          "{mean} ({sd})",
+          "{median} ({min} - {max})"
         ),
-        frmt_structure(
-          group_val = ".default", label_val = "Median (Min - Max)",
-          frmt_combine("{Median} ({Min} - {Max})", Median = frmt("xx.x"), Min = frmt("xx.x"), Max = frmt("xx.x"))
-        )
+        all_categorical() ~ "{n} ({p}%)"
       ),
-      col_plan = col_plan(-starts_with("ord")),
-      big_n = big_n_tbl
+      digits = all_continuous() ~ 2,
+      missing = "no"
     ) %>%
-      print_to_gt(tbl)
+    gtsummary::bold_labels() %>%
+    gtsummary::modify_header(all_stat_cols() ~ "**{level}**  \nN = {n}") %>%
+    gtsummary::add_overall(last = TRUE, col_label = paste0("**", "Total Population", "**  \nN = {n}")) %>%
+    gtsummary::add_stat_label(label = all_continuous2() ~ c("Mean (SD)", "Median (min - max)")) %>%
+    gtsummary::modify_footnote(everything() ~ NA) %>%
+    gtsummary::modify_column_alignment(columns = all_stat_cols(), align = "right")
+
+  if (return_ard) {
+    ard <- gtsummary::gather_ard(tbl)
+    res <- list(tbl = tbl, ard = ard)
+  } else {
+    res <- list(tbl = tbl)
   }
```

Comment:
This will fix the no output that Jessica noticed.
```suggestion
  if (return_ard) {
    ard <- gtsummary::gather_ard(tbl)
    res <- list(tbl = tbl, ard = ard)
  } else {
    res <- list(tbl = tbl)
  }
```